File: back_end/fruitsapp_backend/Models/Order.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace fruitsapp_backend.Models
{
	public class Order
	{
        [Key]
        public int Id { set; get; }
        public int? customer_id { set; get; }
        public string? invoice_code { set; get; }
        public int? total_quantity_product { set; get; }
        public string? total_payment { set; get; } //Tổng giá trị đơn hàng (sau khi áp dụng giảm giá và phí vận chuyển)
        public string? payment_method { set; get; }
        public OrderStatus? status_order { set; get; }
        public string? notes { set; get; }
        public int? shipping_fee { set; get; }
        public string? cancel_reason { set; get; } // Lý do huỷ đơn hàng
        public DateTime create_at { set; get; }
        public DateTime update_at { set; get; }
        public bool isDelete { set; get; }


        public Customer? Customer { set; get; }

    }


    // Định nghĩa enum cho trạng thái của đơn hàng
    public enum OrderStatus
    {
        Pending,    // Đơn hàng đang chờ xử lý
        Processing, // Đơn hàng đang được xử lý
        Shipped,    // Đơn hàng đã được gửi đi
        Completed,  // Đơn hàng đã hoàn thành
        Canceled    // Đơn hàng đã bị hủy
    }
}

File: font_end/fruitsapp_admin/src/Models/Order.js
class Order {
    constructor(id, customer_id, invoice_code, total_quantity_product, total_payment, payment_method, status_order, notes, shipping_fee, cancel_reason, create_at, update_at) {
        this.id = id;
        this.customer_id = customer_id;
        this.invoice_code = invoice_code;
        this.total_quantity_product = total_quantity_product;
        this.total_payment = total_payment;
        this.payment_method = payment_method;
        this.status_order = status_order;
        this.notes = notes;
        this.shipping_fee = shipping_fee;
        this.cancel_reason = cancel_reason;
        this.create_at = create_at;
        this.update_at = update_at;
    }

    // Phương thức để tính tổng tiền sau khi thêm phí vận chuyển
    calculateTotalWithShipping() {
        return this.total_payment + this.shipping_fee;
    }

    // Phương thức để hiển thị thông tin đơn hàng
    getOrderDetails() {
        return `
        Order ID: ${this.id}
        Invoice Code: ${this.invoice_code}
        Customer ID: ${this.customer_id}
        Total Products: ${this.total_quantity_product}
        Total Payment: $${this.total_payment}
        Payment Method: ${this.payment_method}
        Status: ${this.status_order}
        Notes: ${this.notes ? this.notes : 'No notes'}
        Shipping Fee: $${this.shipping_fee}
        Cancel Reason: ${this.cancel_reason ? this.cancel_reason : 'No cancellation'}
        Created At: ${this.create_at}
        Updated At: ${this.update_at}
        `;
    }

    // Phương thức để kiểm tra xem đơn hàng có bị hủy không
    isCancelled() {
        return this.status_order === 'cancelled';
    }
}

export default Order;
